Cesium3DTilesSelection/src/LayerJsonTerrainLoader.cpp

```diff
@@ -727,69 +766,108 @@ Future<TileLoadResult> LayerJsonTerrainLoader::loadTileContent(
+        // need to generate its raster overlay UVs in the worker thread based
+        // on the projection of the loader since the upsampler needs this UV
+        // to do the upsampling
+        auto result = convertToTileLoadResult(std::move(loadResult));
+        if (tileHasUpsampledChild &&
+            result.state == TileLoadResultState::Success) {
+          generateRasterOverlayUVs(
+              tileBoundingVolume,
+              tileTransform,
+              projection,
+              result);
+        }
+
+        return result;
       });
 }
 
-bool LayerJsonTerrainLoader::updateTileContent(Tile& tile) {
+TileChildrenResult
+LayerJsonTerrainLoader::createTileChildren(const Tile& tile) {
   const CesiumGeometry::QuadtreeTileID* pQuadtreeID =
       std::get_if<CesiumGeometry::QuadtreeTileID>(&tile.getTileID());
   if (pQuadtreeID) {
-    if (tile.getChildren().empty()) {
-      // For the tile that is in the middle of subtree, it is safe to create the
-      // children. However for tile that is at the availability level, we have
-      // to wait for the tile to be finished loading, since there are multiple
-      // subtrees in the background layers being on the flight as well. Once the
-      // tile finishes loading, all the subtrees are resolved
-      bool isTileAtAvailabilityLevel = false;
-      for (const auto& layer : _layers) {
-        if (layer.availabilityLevels > 0 &&
-            int32_t(pQuadtreeID->level) % layer.availabilityLevels == 0) {
-          isTileAtAvailabilityLevel = true;
-          break;
-        }
-      }
-
-      if (isTileAtAvailabilityLevel &&
-          tile.getState() <= TileLoadState::ContentLoading) {
-        return true;
+    // For the tile that is in the middle of subtree, it is safe to create the
+    // children. However for tile that is at the availability level, we have
+    // to wait for the tile to be finished loading, since there are multiple
+    // subtrees in the background layers being on the flight as well. Once the
+    // tile finishes loading, all the subtrees are resolved
+    bool isTileAtAvailabilityLevel = false;
+    for (const auto& layer : _layers) {
+      if (layer.availabilityLevels > 0 &&
+          int32_t(pQuadtreeID->level) % layer.availabilityLevels == 0) {
+        isTileAtAvailabilityLevel = true;
+        break;
       }
+    }
 
-      createTileChildren(tile);
-      return false;
+    if (isTileAtAvailabilityLevel &&
+        tile.getState() <= TileLoadState::ContentLoading) {
+      return {{}, TileLoadResultState::RetryLater};
     }
 
-    return true;
+    return {createTileChildrenImpl(tile), TileLoadResultState::Success};
   }
 
-  return false;
+  return {{}, TileLoadResultState::Failed};
 }
 
-void LayerJsonTerrainLoader::createTileChildren(Tile& tile) {
-
-  if (tile.getChildren().empty()) {
+bool LayerJsonTerrainLoader::doesTileHasUpsampledChild(const Tile& tile) const {
```

Comment:
`doesTileHaveUpsampledChild`

or alternatively: `tileHasUpsampledChild` is fine too.